extension/chrome/elements/compose-modules/formatters/encrypted-mail-msg-formatter.ts

```diff
@@ -22,19 +22,77 @@ import { AcctStore } from '../../../../js/common/platform/store/acct-store.js';
+      for (const [key, value] of Object.entries(newMsg.recipients)) {
+        if (['to', 'cc', 'bcc'].includes(key)) {
+          const sendingType = key as RecipientType;
+          pubkeyRecipients[sendingType] = value?.filter(emailPart => pubkeys.some(p => p.email === emailPart.email)
+            // pwd recipients that have no personal links go to legacy message
+            || (uploadedMessageData.emailToExternalIdAndUrl || {})[emailPart.email] === undefined);
+        }
+      }
+      const uniquePubkeyRecipientToAndCCs = Value.arr.unique((pubkeyRecipients.to || []).concat(pubkeyRecipients.cc || [])
+        .map(recipient => recipient.email.toLowerCase()));
+      // pubkey recipients should be able to reply to "to" and "cc" pwd recipients
+      const replyTo = (newMsg.recipients.to ?? []).concat(newMsg.recipients.cc ?? [])
+        .filter(recipient => !uniquePubkeyRecipientToAndCCs.includes(recipient.email.toLowerCase()));
+      const msgs: SendableMsg[] = [];
+      if (pubkeyRecipients.to?.length || pubkeyRecipients.cc?.length || pubkeyRecipients.bcc?.length) {
+        const pubkeyMsgData = {
+          ...newMsg,
+          recipients: pubkeyRecipients,
+          // brackets are required for test emails like '@test:8001'
+          replyTo: replyTo.length ? `${Str.formatEmailList([newMsg.from, ...replyTo], true)}` : undefined
+        };
+        msgs.push(await this.sendablePwdMsg(
+          pubkeyMsgData,
+          pubkeys,
+          { msgUrl: uploadedMessageData.url, externalId: uploadedMessageData.externalId },
+          collectedAttachments,
+          signingKey?.key)
+        );
+      }
+      // adding individual messages for each recipient that doesn't have a pubkey
+      for (const recipientEmail of pwdRecipients) {
+        const { url, externalId } = uploadedMessageData.emailToExternalIdAndUrl![recipientEmail];
+        const foundParsedRecipient = (newMsg.recipients.to ?? []).concat(newMsg.recipients.cc ?? []).concat(newMsg.recipients.bcc ?? []).
+          find(r => r.email.toLowerCase() === recipientEmail.toLowerCase());
+        // todo: since a message is allowed to have only `cc` or `bcc` without `to`, should we preserve the original placement(s) of the recipient?
+        const individualMsgData = { ...newMsg, recipients: { to: [foundParsedRecipient ?? { email: recipientEmail }] } };
+        msgs.push(await this.sendablePwdMsg(individualMsgData, pubkeys, { msgUrl: url, externalId }, [], signingKey?.key));
+      }
+      return { senderKi: signingKey?.keyInfo, msgs, recipients: newMsg.recipients, attachments: collectedAttachments };
+    } else {
+      const msg = await this.sendableNonPwdMsg(newMsg, pubkeys, signingKey?.key);
+      return {
+        senderKi: signingKey?.keyInfo,
+        msgs: [msg],
+        recipients: msg.recipients,
+        attachments: msg.attachments // todo: perhaps, we should hide technical attachments, like `encrypted.asc` and use collectedAttachments too?
```

Comment:
> `sendableNonPwdMsg` decides how to format it based on rich text or simple text email, and I guess adding `msg.attachments` preserves that.

The top-level properties of `MultipleMessages` like `recipients` and `attachments` are only used to render the message after sending in the inbox reply thread (so we don't have to collect attachments and recipients across multiple messages).
Actual attachments to be sent are per-message inside `msgs`.

Reply:
I see. Let's do this `renderSentMessage: {recipients, attachments}` instead of 

```ts
        recipients: msg.recipients,
        attachments: msg.attachments // todo: perhaps, we should hide technical attachments, like `encrypted.asc` and use collectedAttachments too?
```

And then hide technical attachments as you say, in another issue.